src/crimson/net/SocketMessenger.cc

```diff
@@ -15,18 +15,60 @@
 #include "SocketMessenger.h"
 
 #include <tuple>
+#include <boost/functional/hash.hpp>
 
 #include "auth/Auth.h"
 #include "Errors.h"
 #include "Dispatcher.h"
 
 using namespace ceph::net;
 
+namespace {
+  seastar::logger& logger() {
+    return ceph::get_logger(ceph_subsys_ms);
+  }
+}
+
 SocketMessenger::SocketMessenger(const entity_name_t& myname)
-  : Messenger{myname}
+  : Messenger{myname}, sid{seastar::engine().cpu_id()}
 {}
 
-void SocketMessenger::bind(const entity_addr_t& addr)
+seastar::future<> SocketMessenger::bind(const entity_addr_t& addr)
+{
+  return container().invoke_on_all([addr](auto& msgr) {
+      msgr.do_bind(addr);
```

Comment:
👍  so we listen on all shards now.

Reply:
@cyx1231st after a second thought, i think we don't need this, if we are going to re-distribute the connections to all cores.

Reply:
If the seastar-app is started with multi-cores, and single-threaded-osds are running in the different cores of the same seastar-app, we may still need to listen and accept on all cores. It is because the fd will be [moved to different cores unconditionally]( https://github.com/scylladb/seastar/blob/master/src/net/posix-stack.cc#L221) according to its LB strategy.

There are related discussions about it:
* From [seastar-dev](https://groups.google.com/d/topic/seastar-dev/PojoIl9NBhk/discussion):
   ```
   In the Seastar tutorial, "Introducing Seastar's Network Stack" section,
   I discussed this topic: a design flaw (?) of Seastar's listen() API is that
   even if you run it on a single shard (core), Seastar will actually direct
   the incoming connections to all cores, so you are forced to also accept()
   and handle the connections on all cores.
   ```
* From [seastar tutorial](https://github.com/scylladb/seastar/blob/master/doc/tutorial.md#introducing-seastars-network-stack):
   ```
   Unless the server is run with the "-c1" option (one thread only),
   this will mean that any connection arriving to a different thread
   will not be handled. So in all the examples below, we will need to
   run the same service loop on all cores.
   ```